include/fast_float/ascii_number.h

```diff
@@ -86,40 +87,42 @@ fastfloat_really_inline constexpr bool is_made_of_eight_digits_fast(uint64_t val
      0x8080808080808080));
 }
 
+template <typename UC>
 fastfloat_really_inline FASTFLOAT_CONSTEXPR20
-bool is_made_of_eight_digits_fast(const char *chars)  noexcept  {
+bool is_made_of_eight_digits_fast(UC const * chars)  noexcept  {
   return is_made_of_eight_digits_fast(read_u64(chars));
 }
 
-typedef span<const char> byte_span;
-
-struct parsed_number_string {
+template <typename UC>
+struct parsed_number_string_t {
   int64_t exponent{0};
   uint64_t mantissa{0};
-  const char *lastmatch{nullptr};
+  UC const * lastmatch{nullptr};
   bool negative{false};
   bool valid{false};
   bool too_many_digits{false};
   // contains the range of the significant digits
-  byte_span integer{};  // non-nullable
-  byte_span fraction{}; // nullable
+  span<UC> integer{};  // non-nullable
+  span<UC> fraction{}; // nullable
```

Comment:
span\<const UC\> is better here and in other places

Reply:
I have fixed this in a commit.

Reply:
We take as inputs (your code): 
```C++
UC const *p, UC const * pend
```

So we always take pointers to constant memory.

Now consider the code in context...

```C++
   UC const * const start_digits = p; // @Pharago 's code 
  (...)
   answer.integer = span<const UC>(start_digits, size_t(digit_count)); // @lemire added the const
```

The point is moot because we don't actually use `std::span`, and in our version, a span is always pointing a const content. But if we adopted C++20 spans, then the code would not build without the new const. Furthermore, the existing library relies on `const char`.


In any case, I submit to you that it changes nothing practically speaking. I added the const because that's how the code was before and I want this PR to change as few things as possible.